cache/models/samples.js

```diff
@@ -586,7 +596,7 @@ module.exports = {
    */
   bulkUpsertSample(sampleQueryBody) {
     const promises = sampleQueryBody.map(
-      (sampleReq) => upsertOneSample(sampleReq)
+      (sampleReq) => upsertOneSample(sampleReq, null, true)
     );
```

Comment:
I needed to pass isBulk to customize response based on caller method (null is request method to create api links which we don not want in bulk upsert). I did this to match existing results, but I feel we can improve on that. In case of bulk upsert, we are sending this object {isFailed: true, Explanation: [errobject] }, and in simple upsert we just send err object.
I think we should change this behaviour and return error object directly in both cases and check error in caller function (bulk upsert) by using `instanceof Error `
If we do this, we will be changing in non-redis implementation as well (which I didnt wanna touch with this PR)